src/services/users.services.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const { generateId } = require('../helpers/helpers.js');
+
+/**
+ *@typedef {Object} User
+ *@property {number} id
+ *@property {string} name
+ *@type {User[] | []}
+ */
+let users = [];
+
+const getAll = () => {
+  return users.length !== 0 ? users : [];
+};
+
+const getUserById = (id) => {
+  return users.find((user) => user.id === id) || null;
+};
+
+const removeById = (id) => {
+  const newUsers = users.filter((user) => user.id !== id);
+
+  const isRemoved = newUsers.length !== users.length || null;
+
+  users = newUsers;
+
+  return isRemoved;
+};
+
+const createUser = (name) => {
+  const newUser = {
+    id: generateId(),
+    name,
+  };
+
+  users.push(newUser);
+
+  return newUser;
+};
+
+const updateById = (id, name) => {
+  const searchedUser = users.find((user) => user.id === id);
+
+  const result = searchedUser ? Object.assign(searchedUser, { name }) : null;
```

Comment:
If you don't find user you can throw an error and send to the client that user not found